WIP: Use LCP license ID as a loan's external identifier to fetch the current library

vbessonov opened the pull request:
## Description

<!--- Describe your changes -->

This PR updates LCP endpoint URLs. I described the issue in more detail in [SIMPLY-3121](https://jira.nypl.org/browse/SIMPLY-3121).

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

leonardr commented on api/base_controller.py:
```diff
@@ -82,6 +86,38 @@ def authenticate(self):
+                'An unexpected exception occurred during fetching a loan with external identifier {0}'.format(
+                    loan_external_identifier
+                )
+            )
```
This method should either re-throw the exception or return a problem detail document; probably the former since an exception here is most likely a problem with the database.

vbessonov replied:
[Line # 113](https://github.com/NYPL-Simplified/circulation/pull/1495/files/2ab30d75eb96d2ec2fe9b40f6032263fd5c446e3#diff-f258e5927d0cab67f7bc9611720642a5R113) checks whether `loan` is None and returns `LOAN_NOT_FOUND` if it's empty.
Do you want to return a more specific problem detail document in the case of an exception?

leonardr replied:
Normally we wouldn't catch this exception at all -- any database access might raise an unexpected exception. If that happens we would allow it to be treated as an uncaught exception because there's a serious problem with the database. My preferred solution would be to get rid of the try/except block altogether, but if you saw something unusual during development, returning a specific problem detail might be appropriate.

vbessonov replied:
Removed try/except block

leonardr commented on api/routes.py:
```diff
@@ -570,14 +598,16 @@ def saml_callback():
 
-@app.route('/lcp/hint')
+@library_route('/lcp/hint')
+@has_library
```
Presumably you're going to change this later to use `has_library_through_external_loan_identifier`. If you're confident that your LCP branch will be accepted, I think the change can happen now; we already know the code won't work as is.

vbessonov replied:
I updated the `hint` route to `/<collection_name>/lcp/licenses/<license_id>/hint'`

leonardr reviewed (changes requested):
Looks good overall. The only thing I really need to approve is the code shouldn't swallow an exception in get_one.

vbessonov commented:
@leonardr, I updated `lcp/hint` endpoint's URL and added `license_id` parameter to it. It's working locally with the changes I did in [readium-lcp-server](https://github.com/readium/readium-lcp-server/pull/227) but I'll test it one more time once Laurent approves and merges my PR.

vbessonov commented:
@leonardr, Laurent merged my [PR](https://github.com/readium/readium-lcp-server/pull/227), I tested it locally and it worked as expected

leonardr reviewed (approved):
Great, sounds like we're good to go.
